updater/lib/dependabot/dependency_change.rb

```diff
@@ -46,5 +46,51 @@ def updated_dependency_files_hash
     def grouped_update?
       !!dependency_group
     end
+
+    # This method combines checking the job's `updating_a_pull_request` flag
+    # with verification the dependencies involved remain the same.
+    #
+    # If the dependencies involved have changed, we should close the old PR
+    # rather than supersede it as the new changes don't necessarily follow
+    # from the previous ones; dependencies could have been removed from the
+    # project, or pinned by other changes.
+    def should_replace_existing_pr?
+      return false unless job.updating_a_pull_request?
+
+      # NOTE: Gradle, Maven and Nuget dependency names can be case-insensitive
+      # and the dependency name injected from a security advisory often doesn't
+      # match what users have specified in their manifest.
+      updated_dependencies.map(&:name).map(&:downcase) != job.dependencies.map(&:downcase)
+    end
+
+    def matches_existing_pr?
+      !!existing_pull_request
+    end
+
+    private
+
+    def existing_pull_request
+      if grouped_update?
+        # We only want PRs for the same group that have the same versions
+        job.existing_group_pull_requests.find do |pr|
+          pr["dependency-group-name"] == dependency_group.name &&
+            Set.new(pr["dependencies"]) == updated_dependencies_set
+        end
+      else
+        job.existing_pull_requests.find { |pr| Set.new(pr) == updated_dependencies_set }
+      end
```

Comment:
We may want to break out the "existing grouped update PR?" check into its own method that we can call in `#upsert_pull_request`

Users could define a group that has a single dependency and matches an existing ungrouped PR, causing `dependency_change.matches_existing_pr?` to return true.
If that happens, then we would close the existing ungrouped dependency PR in favor of a grouped one (which could be okay?)

If we introduce a `#matches_existing_grouped_pr?` method then we can make sure we only upsert grouped updates in the refresh grouped update PR operation

Reply:
> Users could define a group that has a single dependency and matches an existing ungrouped PR, causing dependency_change.matches_existing_pr? to return true. 

This shouldn't happen at present as `grouped_updated?` on line 73 would be true for the grouped update meaning it wouldn't check the `existing_pull_requests` set for any matches, so it should keep both job types in their own lane.

We might want to consider doing a one-time check between strategies when users change their config to add a group ( i.e. close any old single PRs that match the new group ), but I think we can leave that for now and allow it to be managed manually so it's ok not to do any cross-checking and do a hard fork on `grouped_updated?` being true/false for the current job.